sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -155,9 +155,13 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
     .set(SQLConf.SHUFFLE_PARTITIONS, 4)
 
   /** List of test cases to ignore, in lower cases. */
-  protected def ignoreList: Set[String] = Set(
-    "ignored.sql"   // Do NOT remove this one. It is here to test the ignore functionality.
-  )
+  protected val ignoreList: Set[String] = if (TestUtils.testCommandAvailable("/bin/bash")) {
+    // SPARK-32106 Since we add SQL test 'transform.sql' will use `cat` command,
+    // here we need to ignore it.
+    Set("ignored.sql", "transform.sql")
+  } else {
+    Set("ignored.sql")   // Do NOT remove this one. It is here to test the ignore functionality.
+  }
```

Comment:
```suggestion
  protected def ignoreList: Set[String] = Set(
    "ignored.sql"   // Do NOT remove this one. It is here to test the ignore functionality.
  ) ++ {
    // SPARK-32106 Since we add SQL test 'transform.sql' will use `cat` command,
    // here we need to ignore it.
    if (TestUtils.testCommandAvailable("/bin/bash")) Set("transform.sql") else Nil
  }
```